### PR TITLE
Simplify test results data

### LINKED ISSUE #10: Test results data structure design
This project currently emits data collected from executing tests as JSON-formatted text. I'd like to discuss improvements to the structure and content of those reports.

[Here's what the data structure currently looks like](https://github.com/w3c/aria-at-automation-harness/blob/f8b8114b068e737410025fb3bb9b6861bfd64dcf/src/data/types.js#L64-L83) (expressed using [JSDoc](https://jsdoc.app/)-style JavaScript code comments):

```js
/**
 * Result from a test plan.
 * @typedef AriaATCIData.TestPlanResult
 * @property {string} name name of the test plan, defaults to 'unknown'
 * @property {AriaATCIData.Log[]} log debug messages emitted during execution of test plan
 * @property {object[]} tests
 * @property {string} tests[].filepath filepath of file describing the test in the test plan
 * @property {AriaATCIData.Log[]} tests[].log subset of log emitted during this single test
 * @property {AriaATCIData.TestResult[]} tests[].results
 */

/**
 * Result from a single test in a test plan.
 * @typedef AriaATCIData.TestResult
 * @property {number} testId numeric id of a test in a test plan
 * @property {object[]} results permutation of input commands and assertions passing or not passing
 * @property {string} results[].command id of input command sent to system
 * @property {string} results[].expectation description of expected assertion
 * @property {boolean} results[].pass did command pass or not pass expectation
 */
```

@mzgoddard As described, the two properties named `log` sound like they may be redundant. Could the "debug messages emitted during execution of test plan" be recreated from the "subset of log emitted during [each] single test"?

Some suggestions:

- Lets add timestamps for the moment that testing started and ended
- Because individual tests can be interrupted for any number of exceptional circumstances, and because those circumstances do not necessarily reflect a failure to meet expectations, results should be modeled with at least three states: "pass", "fail", and "error". The "fail" and "error" states generally need to be further contextualized with a description of the failure or exception, warranting an additional property.
- We'll eventually want to include information on the system under test (i.e. the names and versions of the operating system, web browser, and screen reader). While we work out the details of how those values are inferred, it might be good to include null values in the data structure, if only as a signal of intent

### PR DESCRIPTION
**Pull request by ChrisC**
Closes #10 while [also removing some additional IPC cruft](https://github.com/w3c/aria-at-automation-harness/commit/ba4ad0b4e74524af6f2d863111a2fd86019799fc) in the way we were loading the test plans from the test files into the harness. 

There was one request to in #10 to move the testId out of `TestResult` and into the `TestPlanResult`. Unfortunately, I didn't get to this, as it looks to me like it would require a bigger refactor of how [we're loading the test plans via the CLI middleware](https://github.com/bocoup/aria-at-automation-harness/blob/ef300f3fff302ec6be642fddf1a0199343aecfe4/src/host/plan-from.js#L71). We don't actually get the test Id until later in the run process, when we [JSON parse the file to run the tests](https://github.com/w3c/aria-at-automation-harness/blob/main/src/host/main.js#L119). If you have a simple suggestion for making this change as requested, I'm all ears @jugglinmike ...

### CODE REVIEW

**Comment by jugglinmike** [comment]
I just got all sorts of confused because we don't actually *use* the `AriaATCIData.TestPlanResult` type. It looks like [we just emit a modified version of `AriaATCIHost.TestPlan`](https://github.com/w3c/aria-at-automation-harness/blob/eb90d4bc84630fc8398beef572e415c8a346acee/src/host/main.js#L167), instead. Maybe we could extend [the modification logic](https://github.com/w3c/aria-at-automation-harness/blob/eb90d4bc84630fc8398beef572e415c8a346acee/src/host/cli-run-plan.js#L228-L235) because we have the testId at that moment. That doesn't seem like a huge refactoring to me, but maybe I've missed something important. In either case, I'm happy to punt that to another day if you don't mind creating an issue/Task for it.

**Comment by ChrisC** [comment]
@jugglinmike I got that `testId` property update in the last commits. Here's some sample `TestPlanResult` output as of https://github.com/w3c/aria-at-automation-harness/pull/70/commits/c259f16917185f633464014781f2227ad1a25f42:
```
{
    filepath: "test-06-navBackToSlider-voiceover_macos.collected.json",
    id: "navBackToSlider",
    log: [ ... ],
    results: [
      {
        capabilities: { ... },
        commands: [
          {
            command: "ctrl+opt+left ctrl+opt+left",
            response: "128\n            Red             slider\nRed\nYou are currently on a selectable text. ",
            assertions: [
              {
                expectation: "Role 'slider' is conveyed",
                verdict: null,
              },
              {
                expectation: "Name 'Red' is conveyed",
                verdict: null,
              },
              {
                expectation: "Value '128' is conveyed",
                verdict: null,
              },
              {
                expectation: "Orientation 'horizontal' is conveyed",
                verdict: null,
              },
              {
                expectation: "Minimum value '0' is conveyed",
                verdict: null,
              },
              {
                expectation: "Maximum value '255' is conveyed",
                verdict: null,
              },
              {
                expectation: "Screen reader switched from reading mode to interaction mode",
                verdict: null,
              },
            ],
          },
          { ... },
          { ... },
        ],
      },
    ],
  },
```

File: src/data/types.js [hunk 1]
```diff
@@ -70,15 +70,30 @@
...
- * @property {AriaATCIData.TestResult[]} tests[].results
+ * @property {AriaATCIData.TestResultOutput[]} tests[].results
  */
 
 /**
  * Result from a single test in a test plan.
  * @typedef AriaATCIData.TestResult
- * @property {number} testId numeric id of a test in a test plan
+ * @property {string} testId id of a test in a test plan
+ * @property {number} presentationNumber numeric id of a test in a test plan
```
Comment (ChrisC) [inline src/data/types.js:83]: Note that in the previous version we were inserting the string id as `testId` even though the type  specifies that we were expecting a numeric id. I updated the types to reflect this. I'm assuming we want to stick with using the string version as the `id` and not the numeric `presentationNumber`...

**Comment by jugglinmike** [comment]
The code looks good, and it functions as expected. Thanks for doing this, @ChrisC--these patches are really making the code base smaller and more coherent!

**Pull request merged**